sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveHintsSuite.scala

```diff
@@ -241,4 +241,52 @@ class ResolveHintsSuite extends AnalysisTest {
       Project(testRelation.output, testRelation),
       caseSensitive = false)
   }
+
+  test("Supports multi-part table names for broadcast hint resolution") {
+    // local temp table (single-part identifier case)
+    checkAnalysis(
+      UnresolvedHint("MAPJOIN", Seq("table", "table2"),
+        table("table").join(table("table2"))),
```

Comment:
The following will be better because this is a `caseSensitive = false` test case.
```scala
- table("table").join(table("table2"))),
+ table("TaBlE").join(table("TaBlE2"))),
```

Reply:
Yea, yes!